keyboards/primekb/prime_e/keymaps/jetpacktuxedo/keymap.c

```diff
@@ -39,41 +51,72 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
   [3] = LAYOUT(
     KC_TRNS, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,       KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
-    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_CAPS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
     KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,              KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS
   )
 };
 
 void matrix_init_user(void) {
   // set CapsLock LED to output and low
-  setPinOutput(B1);
-  writePinLow(B1);
+  setPinOutput(CAPS_LED_PIN);
+  writePinLow(CAPS_LED_PIN);
   // set NumLock LED to output and low
-  setPinOutput(B2);
-  writePinLow(B2);
+  setPinOutput(NUM_LED_PIN);
+  writePinLow(NUM_LED_PIN);
   // set ScrollLock LED to output and low
-  setPinOutput(B3);
-  writePinLow(B3);
+  setPinOutput(SCROLL_LED_PIN);
+  writePinLow(SCROLL_LED_PIN);
+}
+
+void matrix_scan_user(void) {
+    if (CAPS == 1) {
+        // Blink the first led when capslock is active
+        if (BLINK_TIMER >= BLINK_DURATION) {
+            togg_indicator(&CAPS_LED_STATE, CAPS_LED_PIN);
+            BLINK_TIMER = 0;
+        }
+        BLINK_TIMER++;
+    }
 }
 
 //function for layer indicator LED
-uint32_t layer_state_set_user(uint32_t state)
-{
-    if (state & (1<<1)) {
-    writePinHigh(B1);
+uint32_t layer_state_set_user(uint32_t state) {
```

Comment:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

Reply:
Sure can, didn't realize we had a new type for that now. I'll update that in just a few.

Reply:
Oops, didn't realize I could have just directly committed your suggestion. Updated now though.

Reply:
Yeah, it was added a few months ago.  It makes things nice, if you want to reduce the number of max layers. :)

And yeah, the "commit suggestion" is super nice!